Buildings/Fluid/MixingVolumes/MixingVolumeEvaporation.mo

```diff
@@ -26,53 +28,157 @@ model MixingVolumeEvaporation
         rotation=0,
         origin={110,70})));
 
+  // Parameter
+  parameter Modelica.SIunits.Volume V "Total volume";
+
+// Variables
+  MediumWat.ThermodynamicState state_saturated
+    "Saturated state";
+  MediumSte.AbsolutePressure p(
+    start=p_start,
+    stateSelect=if massDynamics == Modelica.Fluid.Types.Dynamics.SteadyState
+    then StateSelect.default else StateSelect.prefer)
+    "Pressure inside volume";
+  MediumSte.Temperature T(start=T_start) "Temperature inside volume";
+  Modelica.SIunits.Volume VSte(final start=VSte_start) "Volume of steam vapor";
+  Modelica.SIunits.Volume VWat(
+    start=VWat_start,
+    stateSelect=if massDynamics == Modelica.Fluid.Types.Dynamics.SteadyState
+    then StateSelect.default else StateSelect.prefer)
+    "Volume of liquid water phase";
+  MediumSte.SpecificEnthalpy hSte "Specific enthalpy of steam vapor";
+  MediumWat.SpecificEnthalpy hWat "Specific enthalpy of liquid water";
+  MediumSte.Density rhoSte "Density of steam vapor";
+  MediumWat.Density rhoWat "Density of liquid water";
+  Modelica.SIunits.Mass m "Total mass of volume";
+  Modelica.SIunits.Energy U(final start=U_start) "Internal energy";
+  Modelica.SIunits.MassFlowRate mWat_flow "Feed water mass flow rate";
+  Modelica.SIunits.MassFlowRate mSte_flow "Steam mass flow rate";
+
+  // Initialization
+  parameter Modelica.SIunits.Volume VWat_start=V/2
+    "Start value of liquid volume"
+    annotation (Dialog(tab="Initialization"));
+  final parameter Modelica.SIunits.Volume VSte_start= V-VWat_start
+  "Start value of steam vapor volume";
+  final parameter MediumWat.ThermodynamicState state_start = MediumWat.setState_pTX(
+      T=T_start,
+      p=p_start,
+      X=MediumWat.X_default) "Medium state at default values";
+  final parameter Modelica.SIunits.Density rhoWat_start=MediumWat.density(
+    state=state_start) "Density, used to compute fluid mass";
+  final parameter Modelica.SIunits.Density rhoSte_start=MediumSte.density(
+    state=state_start) "Density, used to compute fluid mass";
+  final parameter Modelica.SIunits.SpecificEnthalpy hSteStart=
+    MediumSte.specificEnthalpy_pTX(p_start, T_start, MediumSte.X_default)
+    "Start value for specific enthalpy";
+  final parameter Modelica.SIunits.Energy U_start=
+    VWat_start*rhoWat_start*MediumWat.specificEnthalpy(state_start) +
+    VSte_start*rhoSte_start*MediumSte.specificEnthalpy(state_start) -
+    p_start*V
+    "Starting internal energy";
+
```

Comment:
So I revisited this issue this morning, and talked with Michael about the distinction between `start` attribute and initial equation, and I got a solution that might work.
First, all of those start variable can be deleted (from `VSte_start` down to `U_start`). They all specify the same calculation as the general calculation and their value can be back calculated from it.
Second, all of the start value will be properly calculated if you specify that the start condition for `VWat` is fixed `VWat(start=VWat_start, fixed=true)`. What this does is change the behavior of `start` from a guess of the initial condition to a fixed initial condition (equivalent to having `VWat = VWat_start` in initial equation). You'll also need to remove the assignment `VWat = VWat_start` in the initial equation of the `FixedInitial` case.
Third, the assignment of `U` in the `massDynamics==SteadyState` case can be refactored from `U = U_start` to `der(U) = 0`.

What this achieve is removing the redundancy of calculating the start value separately and let Modelica naturally propagate their value from the initial condition of pressure and volume, which are sufficient to determine the initial state of all variables.

Reply:
Thanks @bravache for the information! This is good to know. We will make the suggested changes and I'll let you know when this is complete on my branch.